test/functional/Java17andUp/src_170/org/openj9/test/jep389/upcall/UpcallMHWithMixedSigStruTests.java

```diff
@@ -859,11 +859,13 @@ public void test_addDoubleAndFloatDoubleFloatFromStructByUpcallMH() throws Throw
 
 	@Test
 	public void test_addDoubleAndIntDoubleLongFromStructByUpcallMH() throws Throwable {
-		/* The size of [int, double, long] on AIX/PPC 64-bit is 20 bytes without padding by default
-		 * while the same struct is 24 bytes with padding on other platforms.
+		/* The padding in the struct [int, double, long] on AIX/PPC 64-bit is different from
+		 * other platforms as follows:
+		 * 1) there is no padding between int and double.
+		 * 2) there is a 4-byte padding between double and long.
 		 */
 		GroupLayout structLayout = isAixOS ? MemoryLayout.structLayout(C_INT.withName("elem1"), C_DOUBLE.withName("elem2"),
-								longLayout.withName("elem3").withBitAlignment(32)) : MemoryLayout.structLayout(C_INT.withName("elem1"),
+								MemoryLayout.paddingLayout(32), longLayout.withName("elem3")) : MemoryLayout.structLayout(C_INT.withName("elem1"),
```

Comment:
I don't think it works as it doesn't include the difference of padding required on different platforms
```
AIX: [int, double, 4-byte padding, long] <----- a 4-byte padding required between double and long
others: [int, 4-byte padding, double, long] <----- a 4-byte padding required between int and double
```
in which the padding is required & checked in OpenJDK.

Reply:
Doesn't `withBitAlignment()` ensure sufficient (preceding) padding to achieve the requested alignment?

Reply:
`withBitAlignment()` is not intended for the padding job in which case the padding must be explicitly specified in the layout for alignment when required as clarified at https://docs.oracle.com/en/java/javase/17/docs/api/jdk.incubator.foreign/jdk/incubator/foreign/MemoryLayout.html
```
A memory layout can be used to describe the contents of a memory segment in a language neutral fashion. 
There are two leaves in the layout hierarchy,...padding layouts which are used, as the name suggests, 
to represent  a portion of a memory segment which are primarily present for alignment reasons
(see paddingLayout(long)).
```
otherwise, the mismatch in alignment will be captured as exception in OpenJDK regardless of platforms.
e.g. for the layout of the struct `{int a; double b; long c; }`  in JDK20 on `Linux/x86_64`:
```
		GroupLayout structLayout =  MemoryLayout.structLayout(JAVA_INT.withName("elem1"),
						JAVA_DOUBLE.withName("elem2"), JAVA_LONG.withName("elem3")); <--- the alignment of JAVA_LONG on X86_64 is 64 by default.
or 
		GroupLayout structLayout =  MemoryLayout.structLayout(JAVA_INT.withName("elem1"),
						JAVA_DOUBLE.withName("elem2"), JAVA_LONG.withName("elem3").withBitAlignment(64));
```
Without the explicit padding between `int` and `double`, it will end up with `UnsupportedOperationException` for the missing padding before `double` as follows:
```
Exception in thread "main" java.lang.UnsupportedOperationException: Invalid alignment requirements for layout d64(elem2)
        at java.base/jdk.internal.foreign.LayoutPath.checkAlignment(LayoutPath.java:236)
        at java.base/jdk.internal.foreign.LayoutPath.dereferenceHandle(LayoutPath.java:150)
        at java.base/java.lang.foreign.MemoryLayout.computePathOp(MemoryLayout.java:465)
        at java.base/java.lang.foreign.MemoryLayout.varHandle(MemoryLayout.java:394)
        at UpCallTests.test_addDoubleAndIntDoubleLongFromStructByUpcallMH(UpCallTests.java:249)
```